### PR TITLE
frame-benchmarking-cli: Remove native dispatch requirement

### PR DESCRIPTION
**Pull request by bkchr**
No need for this, we can just use the `WasmExecutor` directly.

# Changes

Removes the requirement to pass the `ExecDispatch` generic parameter and instead to pass the host functions directly. If you don't want to add any extra host functions, you can just pass `()` to `PalletCmd::run`.

polkadot companion: https://github.com/paritytech/polkadot/pull/7434
cumulus companion: https://github.com/paritytech/cumulus/pull/2792

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3084034

**Review by ggwpez — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3084271

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3085662

File: frame/contracts/src/schedule.rs [hunk 1]
```diff
@@ -145,21 +145,20 @@ impl Limits {
...
+/// 1. If there is a i64 and a i32 variant of an instruction we use the weight of the former for
+///    both.
+/// 2. The following instructions are free of charge because they merely structure the wasm module
+///    and cannot be spammed without making the module invalid (and rejected): End, Unreachable,
+///    Return, Else
+/// 3. The following instructions cannot be benchmarked because they are removed by any real world
+///    execution engine as a preprocessing step and therefore don't yield a meaningful benchmark
+///    result. However, in contrast to the instructions mentioned in 2. they can be spammed. We
+///    price them with the same weight as the "default" instruction (i64.const): Block, Loop, Nop
+/// 4. We price both i64.const and drop as InstructionWeights.i64const / 2. The reason for that is
```
Comment (sam0x17) [inline frame/contracts/src/schedule.rs:157]:
```suggestion
/// 4. We price both `i64.const` and drop as `InstructionWeights.i64.const` / 2. The reason for that is
```

File: frame/contracts/src/schedule.rs [hunk 2]
```diff
@@ -145,21 +145,20 @@ impl Limits {
...
-///    and dropping return values in order to maintain a valid module.
+/// 1. If there is a i64 and a i32 variant of an instruction we use the weight of the former for
+///    both.
+/// 2. The following instructions are free of charge because they merely structure the wasm module
+///    and cannot be spammed without making the module invalid (and rejected): End, Unreachable,
+///    Return, Else
+/// 3. The following instructions cannot be benchmarked because they are removed by any real world
+///    execution engine as a preprocessing step and therefore don't yield a meaningful benchmark
+///    result. However, in contrast to the instructions mentioned in 2. they can be spammed. We
+///    price them with the same weight as the "default" instruction (i64.const): Block, Loop, Nop
```
Comment (sam0x17) [inline frame/contracts/src/schedule.rs:156]:
```suggestion
///    price them with the same weight as the "default" instruction (`i64.const`): Block, Loop, Nop
```

**Review by sam0x17 — APPROVED** [review]
LGTM, few minor doc tweaks

**Review by skunert — APPROVED** [review]

**Review by davxy — APPROVED** [review]

**Pull request merged**